Source/rclUE/Public/ROS2Publisher.h

```diff
@@ -133,6 +135,33 @@ class RCLUE_API UROS2Publisher : public UROS2Topic
         }
     }
 
+    /**
+     * @brief Publish a message, sends the simulationState
+     *
+     * @tparam TMessageStruct
+     * @param InMessage
+     * @param InSimState
+     */
+    template<typename TUEMessage, typename TUEStruct>
+    void PublishWithSimState(const TUEStruct& InMessageData, ASimulationState* InSimState = nullptr)
+    {
+        // can't use macro in here.
+        // if (IsValid(OwnerNode) && OwnerNode->State == UROS2State::Initialized && State == UROS2State::Initialized)
+        if (State == UROS2State::Initialized)
+        {
+            // Update [TopicMessage] with [InMessageData]
+            CastChecked<TUEMessage>(TopicMessage)->SetMsg(InMessageData);
+            //CastChecked<TUEMessage>(TopicMessage)->SetMsg(InMessageData, InSimState);
+
+            // Publish [TopicMessage]
+            Publish();
+        }
+        else
+        {
+            UE_LOG_WITH_INFO(LogTemp, Log, TEXT("Publish() [%s] Publisher or node is not yet initialized"), *GetName());
+        }
+    }
+

```

Comment:
> Is This function experimental? since it seems there is no real use of `InSimState`, also `ASimulationState` is from RSPlugins, which rclUE is not supposed to depend on

It should be using the commented function :
//CastChecked<TUEMessage>(TopicMessage)->SetMsg(InMessageData, InSimState);
@tetorea can you answer to @Tadinu about the ASimulationState ?

Reply:
@Tadinu oh right, I implemented this to add the char array received in a ros msg inside a globally available array (to handle japanese tote type name encoding issues).
But I'll try to use something else than the simulation state then.